spec/queue.js

```diff
@@ -0,0 +1,76 @@
+import chai, { expect, assert } from 'chai'
+import chaiChange from 'chai-change'
+import Queue from '../src/queue'
+
+chai.use(chaiChange)
+
+describe('Queue', () => {
+  'use strict'
+  let myQueue
+  beforeEach( () => {
+    myQueue = new Queue()
+  })
+
+  it('exists', () => {
+    expect(Queue).to.be.a('function')
+  })
+
+  context('#enqueue(element)', () => {
+    it('should add an element to the back of the queue', () => {
+      expect(() => myQueue.enqueue("foo")).to.alter(() => myQueue.length(), { from: 0, to: 1 })
+    })
+  })
+
+  context('#dequeue', () => {
+    it('should remove and return the front element in the queue', () => {
+      myQueue.enqueue("foo")
+      myQueue.enqueue("bar")
+      expect(myQueue.dequeue()).to.eql("foo")
+      expect(() => myQueue.dequeue()).to.alter(() => myQueue.length(), { from: 1, to: 0 })
+    })
+    it('should return null if queue is empty', () => {
+      expect(myQueue.dequeue()).to.eql(null)
+    })
+  })
+
+  context('#front()', () => {
+    it('should return the front element of the queue', () => {
+      myQueue.enqueue("foo")
+      expect(myQueue.front()).to.eql("foo")
```

Comment:
Just want to point out that this test could be misleading - since there's only one element in the queue under test, it's possible we have a bug that we wouldn't see, since there's only one element that could possibly be returned. I would consider adding another element, just to ensure the *correct* element is getting returned.